### DISCUSSION TITLE
Achieving Uniform Grain Sizes in Neper's Multiscale Tessellations Mode

### DISCUSSION THREAD

**Discussion by PengWei97** [neperfepx/neper#700] [Q&A]
Dear Neper experts,

I am currently using Neper's `multiscale tessellations` mode to generate polycrystalline structures #697. However, I have encountered some challenges in achieving uniform grain sizes. I would greatly appreciate your insights and suggestions on how to address this issue.

Specifically, in the microstructure I generated using the following command, the grains indicate that the distribution is not very uniform，
```bash
neper -T -dim 2 -n "10::msfile(n_msfile)" -id 40 -morpho "gg::msfile(morpho_msfile)" -domain "square(10,10)" -o d -ori 
```
![image](https://github.com/neperfepx/neper/assets/56183775/f24e3dd7-37d1-464b-ac6a-6d95d575dd2f)

Thank you all in advance for your time and assistance!

Best regards,
Wei

**Comment by rquey** [comment]
It is because of `gg`. Change it to specify a custom `diameq` distribution (and add `1-sphericity`):
```
neper -T -dim 2 -n "10::from_morpho" -id 40 -morpho "diameq:lognormal(1,0.01),1-sphericity:lognormal(0.1,0.03)::lamellar(w=0.25)" -domain "square(10,10)" -o d
```
(changed `ms_file` into a standard lamellar tessellation for the example.)
![image](https://github.com/neperfepx/neper/assets/2659376/59179d22-3a9e-46bb-92cf-4af0d0a7face)
